provision/kubernetes/provisioner.go

```diff
@@ -1313,3 +1314,63 @@ func appCustomResourceDefinition() *v1beta1.CustomResourceDefinition {
 		},
 	}
 }
+
+func EnvsForApp(a provision.App, process, imageName string, isDeploy bool) []bind.EnvVar {
+	envs := provision.EnvsForApp(a, process, isDeploy)
+	if isDeploy {
+		return envs
+	}
+
+	yamlData, err := image.GetImageTsuruYamlData(imageName)
+	if err != nil {
+		return append(envs, provision.DefaultWebPortEnvs()...)
+	}
+	portsConfig, err := getProcessPortsForImage(imageName, yamlData, process)
+	if err != nil || len(portsConfig) == 0 {
+		return removeDefaultPortEnvs(envs)
```

Comment:
Should the default env really be removed when loading an invalid config?

Reply:
That makes sense. I'll change to remove the default envs only when `err == nil && len(portsConfig) == 0` (no ports exposed for this process).